cni/network/network.go

```diff
@@ -573,12 +573,20 @@ func (plugin *NetPlugin) cleanupAllocationOnError(
 	options map[string]interface{},
 ) {
 	if result != nil && len(result.IPs) > 0 {
-		if er := plugin.ipamInvoker.Delete(&result.IPs[0].Address, nwCfg, args, options); er != nil {
+		addresses := []*net.IPNet{}
```

Comment:
I believe it's more optimal to make the slice with the length and add the elements using index instead of append

Reply:
If you know the size of the slice ahead of time, then yeah, it's better to `make` it. You can use the 3-argument form to still use `append` if it ends up being cleaner (sometimes juggling indices can just add noise):

```go
addresses := make([]*net.IPNet, 0, len(somethingYouKnow))
```

That creates a 0-length slice with a capacity of `len(somethingYouKnow)`. Because the `len` is 0, `append` still works as expected. But, because the 3rd argument is set, no resizing and copying occurs.